src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java

```diff
@@ -520,6 +520,156 @@ public void xmlRouteBuilder_RestReportIdWorkloadInventory_IdParamGiven_ShouldCal
+
+        StringBuilder sb = new StringBuilder("")
+                .append("orderBy={orderBy}&")
+                .append("orderAsc={orderAsc}&")
+                .append("provider={provider1}&")
+                .append("provider={provider2}&")
+                .append("cluster={cluster1}&")
+                .append("cluster={cluster2}&")
+                .append("datacenter={datacenter1}&")
+                .append("datacenter={datacenter2}&")
+                .append("vmName={vmName1}&")
+                .append("vmName={vmName2}&")
+                .append("osName={osName1}&")
+                .append("osName={osName2}&")
+                .append("workload={workload1}&")
+                .append("workload={workload2}&")
+                .append("recommendedTargetIMS={recommendedTarget1}&")
+                .append("recommendedTargetIMS={recommendedTarget2}&")
+                .append("flagIMS={flag1}&")
+                .append("flagIMS={flag2}&")
+                .append("complexity={complexity1}&")
+                .append("complexity={complexity2}");
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(TestUtil.HEADER_RH_IDENTITY, TestUtil.getBase64RHIdentity());
+        HttpEntity<String> entity = new HttpEntity<>(null, headers);
+
+        //When
+        camelContext.start();
+        TestUtil.startUsernameRoutes(camelContext);
+        camelContext.startRoute("to-sortBean");
+        camelContext.startRoute("to-workloadInventoryFilterBean");
+        camelContext.startRoute("filtered-workload-inventory-report-get-details-as-csv");
+        camelContext.startRoute("workload-inventory-report-model-to-csv");
+
+        ResponseEntity<String> response = restTemplate.exchange(camel_context + "report/{id}/workload-inventory/filtered-csv?" + sb.toString(), HttpMethod.GET, entity, String.class, variables);
+
+        //Then
+        SortBean sortBean = new SortBean(orderBy, orderAsc);
+
+        WorkloadInventoryFilterBean filterBean = new WorkloadInventoryFilterBean();
+        filterBean.setProviders(new HashSet<>(Arrays.asList(provider1, provider2)));
+        filterBean.setClusters(new HashSet<>(Arrays.asList(cluster1, cluster2)));
+        filterBean.setDatacenters(new HashSet<>(Arrays.asList(datacenter1, datacenter2)));;
+        filterBean.setVmNames(new HashSet<>(Arrays.asList(vmName1, vmName2)));;
+        filterBean.setOsNames(new HashSet<>(Arrays.asList(osName1, osName2)));;
+        filterBean.setWorkloads(new HashSet<>(Arrays.asList(workload1, workload2)));
+        filterBean.setRecommendedTargetsIMS(new HashSet<>(Arrays.asList(recommendedTarget1, recommendedTarget2)));
+        filterBean.setFlagsIMS(new HashSet<>(Arrays.asList(flag1, flag2)));
+        filterBean.setComplexities(new HashSet<>(Arrays.asList(complexity1, complexity2)));
+
+        verify(workloadInventoryReportService).findByAnalysisOwnerAndAnalysisId("mrizzi@redhat.com", one, sortBean, filterBean);
```

Comment:
We could take advantage that we have the Spring context up and H2 and check that really the values sorted by Hibernate are the ones we expect.

Reply:
@jonathanvila now the tests are is including insertions in the DB and use it in the asserts